staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -227,10 +199,34 @@ func (s *store) conditionalDelete(ctx context.Context, key string, out runtime.O
 	}
 	for {
 		origState, err := s.getState(getResp, key, v, false)
-		if err != nil {
+		if err != nil && !storage.IsTransformerError(err) {
 			return err
 		}
-		if err := preconditions.Check(key, origState.obj); err != nil {
+		if err != nil && storage.IsTransformerError(err) && preconditions != nil {
+			return err
+		}
+		if err != nil && storage.IsTransformerError(err) && preconditions == nil {
+			// in this rare case, we skip checking
+			// validateDeletion, to give user a way to delete the
+			// key via the apiserver in case of data corruption.
+			startTime := time.Now()
+			// We don't check the transaction response as we know
+			// the transformer cannot interpret it anyway.
+			_, txnErr := s.client.KV.Txn(ctx).If().Then(
+				clientv3.OpDelete(key),
+			).Commit()
+			metrics.RecordEtcdRequestLatency("delete", getTypeName(out), startTime)
+			if txnErr != nil {
+				return txnErr
+			}
+			return err
```

Comment:
if we got a decode error, then deleted unconditionally, and that delete succeeded, I wouldn't expect an error to be returned

Reply:
The original code behaved the same. If you feel strongly, I'd prefer fix it in another pull.